ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -747,4 +750,18 @@ default List<Partition> getPartitionsByExpr(org.apache.hadoop.hive.ql.metadata.T
     throw new UnsupportedOperationException("Storage handler does not support getting partitions by expression " +
             "for a table.");
   }
+
+  default boolean supportsMergeFiles() {
+    return false;
+  }
+
+  default List<FileStatus> getMergeInputFiles(Properties properties) throws IOException {
+    throw new UnsupportedOperationException("Storage handler does not support getting merge input files " +
+            "for a table.");
+  }
+
+  default StorageHandlerMergeProperties getStorageHandlerMergeProperties(Properties properties) {
```

Comment:
not sure about that class, could we get same information from StorageDescriptor?

Reply:
Renamed it to `getMergeProperties`.